internal/datacoord/task_analyze.go

```diff
@@ -182,7 +182,9 @@ func (at *analyzeTask) AssignTask(ctx context.Context, client types.IndexNodeCli
 	totalSegmentsRawDataSize := float64(totalSegmentsRows) * float64(dim) * typeutil.VectorTypeSize(t.FieldType) // Byte
 	numClusters := int64(math.Ceil(totalSegmentsRawDataSize / float64(Params.DataCoordCfg.ClusteringCompactionPreferSegmentSize.GetAsSize())))
 	if numClusters < Params.DataCoordCfg.ClusteringCompactionMinCentroidsNum.GetAsInt64() {
-		numClusters = Params.DataCoordCfg.ClusteringCompactionMinCentroidsNum.GetAsInt64()
+		log.Ctx(ctx).Info("data size is too small, skip analyze task", zap.Float64("raw data size", totalSegmentsRawDataSize), zap.Int64("num clusters", numClusters), zap.Int64("expected num clusters", Params.DataCoordCfg.ClusteringCompactionMinCentroidsNum.GetAsInt64()))
+		at.SetState(indexpb.JobState_JobStateFinished, "")
+		return true, false
```

Comment:
return true, true

first `true` means no error occurred
second `true` means skip the next steps.

Reply:
updated